src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ConfigureCompatibilityOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    public abstract class ConfigureCompatibilityOptions<TOptions> : IPostConfigureOptions<TOptions>
+        where TOptions : class, IEnumerable<ICompatibilitySwitch>
+    {
+        private readonly ILogger _logger;
+
+        internal ConfigureCompatibilityOptions(
+            ILoggerFactory loggerFactory,
+            IOptions<MvcCompatibilityOptions> compatibilityOptions)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            Version = compatibilityOptions.Value.CompatibilityVersion;
+            _logger = loggerFactory.CreateLogger<TOptions>();
+        }
+
+        protected abstract IReadOnlyDictionary<string, object> DefaultValues { get; }
+
+        protected CompatibilityVersion Version { get; }
+
+        public virtual void PostConfigure(string name, TOptions options)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            // Evaluate DefaultValues onces to subclasses don't have to cache.
+            var defaultValues = DefaultValues;
+
+            foreach (var @switch in options)
```

Comment:
Back to actual nits, but if this were named `compatSwitch` everywhere here, it might be easier to read? All this stuff is private/local anyway.

Reply:
Why avoid usage of a language feature for its intended purpose?

Reply:
Meh.